### ISSUE TITLE
Customize settings via webserver and command (settingsUpdate)

### ISSUE THREAD

**Issue by emilianya** [VukkyLtd/VukkyBot#50]
This would make changing config much easier. There would be support for environments where editing local files isnt possible via storing the settings remotely. I am also considering an option to enable a webserver for changing the settings with nice little clicky switches instead of annoying commands!

**Comment by emilianya** [comment]
This is mostly finished IN UNCOMMITTED CHANGES AAAAAAAAAAAAa:
Currently v!settings needs some slight polish
Webserver part is mostly finished, it only needs some logic and I need to get it to talk to the new config API correctly
Config API is done
I need to add an option to disable webserver into config.

**Comment by emilianya** [comment]
This has taken much longer than expected

**Comment by emilianya** [comment]
No progress in 18 days lmao i shoudl finish this

**Comment by emilianya** [comment]
Problem List:

1. when remoteSettings is false: config.js never writes the changed config
2. when remoteSettings is false: config.js never reads the config as seen if you edit it manually
3. when remoteSettings is true: config.js writes multiple thingies for it probably due to an oversight

**Comment by ProFireDev** [comment]
> Problem List:
> 
> 1. when remoteSettings is false: config.js never writes the changed config
> 2. when remoteSettings is false: config.js never reads the config as seen if you edit it manually
> 3. when remoteSettings is true: config.js writes multiple thingies for it probably due to an oversight

where did you guys get to yesterday? did you fix some of the other major issues?

**Comment by emilianya** [comment]
> > Problem List:
> > 
> > 1. when remoteSettings is false: config.js never writes the changed config
> > 2. when remoteSettings is false: config.js never reads the config as seen if you edit it manually
> > 3. when remoteSettings is true: config.js writes multiple thingies for it probably due to an oversight
> 
> where did you guys get to yesterday? did you fix some of the other major issues?

I dont think we did anything

**Comment by ProFireDev** [comment]
Lol I thought you did but maybe that was all front end stuff. It's getting there though

**Comment by emilianya** [comment]
> Lol I thought you did but maybe that was all front end stuff. It's getting there though

Yeah, it was all just faffing on with the 403 page and the buttons

**Comment by ProFireDev** [comment]
> > Lol I thought you did but maybe that was all front end stuff. It's getting there though
> 
> Yeah, it was all just faffing on with the 403 page and the buttons

Oh that's not bad at all then. ~~Basically prod ready~~ maybe not quite lmao. There's some work left still

**Comment by emilianya** [comment]
> > > Lol I thought you did but maybe that was all front end stuff. It's getting there though
> > 
> > 
> > Yeah, it was all just faffing on with the 403 page and the buttons
> 
> Oh that's not bad at all then. ~Basically prod ready~ maybe not quite lmao. There's some work left still

I would much more rather getting the big issues done first, everything else is small tweaks

**Comment by ProFireDev** [comment]
> > > > Lol I thought you did but maybe that was all front end stuff. It's getting there though
> > > 
> > > 
> > > Yeah, it was all just faffing on with the 403 page and the buttons
> > 
> > 
> > Oh that's not bad at all then. ~Basically prod ready~ maybe not quite lmao. There's some work left still
> 
> I would much more rather getting the big issues done first, everything else is small tweaks

For sure, luckly there's not too many big things left to do

**Comment by emilianya** [comment]
> > > > > Lol I thought you did but maybe that was all front end stuff. It's getting there though
> > > > 
> > > > 
> > > > Yeah, it was all just faffing on with the 403 page and the buttons
> > > 
> > > 
> > > Oh that's not bad at all then. ~Basically prod ready~ maybe not quite lmao. There's some work left still
> > 
> > 
> > I would much more rather getting the big issues done first, everything else is small tweaks
> 
> For sure, luckly there's not too many big things left to do

Well its the stuff i have no clue how to do, except 3.

**Comment by ProFireDev** [comment]
Work on #3, then I'll take a look at the best and see what i can do. But google is your friend here. Or maybe even ask a question on stack overflow. But I got lots of free time coming up, so I'll do my best to look into it

**Comment by ProFireDev** [comment]
> Work on #3, then I'll take a look at the best and see what i can do. But google is your friend here. Or maybe even ask a question on stack overflow. But I got lots of free time coming up, so I'll do my best to look into it

But just yolo and figure it out as you go I guess lmao.

That's the best way to learn, and we can go back and fix it later

**Comment by emilianya** [comment]
> > Work on #3, then I'll take a look at the best and see what i can do. But google is your friend here. Or maybe even ask a question on stack overflow. But I got lots of free time coming up, so I'll do my best to look into it
> 
> But just yolo and figure it out as you go I guess lmao.
> 
> That's the best way to learn, and we can go back and fix it later

Wel a 9 year old stackoverflow solution let me down so kinda 😔

**Comment by ProFireDev** [comment]
> > > Work on #3, then I'll take a look at the best and see what i can do. But google is your friend here. Or maybe even ask a question on stack overflow. But I got lots of free time coming up, so I'll do my best to look into it
> > 
> > 
> > But just yolo and figure it out as you go I guess lmao.
> > That's the best way to learn, and we can go back and fix it later
> 
> Wel a 9 year old stackoverflow solution let me down so kinda 😔

lmao of Couse it did, smh at least you didn't find marked as duplicate or stupid question as the answer XD

**Comment by emilianya** [comment]
At this point I would like to leave a message here:
NodeJS on my machine became very drunk and it thinks files contain stuff they dont.
I am unable to work on this

**Comment by ProFireDev** [comment]
> At this point I would like to leave a message here:
> NodeJS on my machine became very drunk and it thinks files contain stuff they dont.
> I am unable to work on this

time to reinstall or something, weird

**Comment by nyakase** [comment]
priority: do when bored

**Comment by ProFireDev** [comment]
> priority: do when bored

it might be a good idea to bump it up to high.... its one of the last things left
